Source/WTF/wtf/Algorithms.h

```diff
@@ -54,5 +58,40 @@ bool allOf(ContainerType&& container, AllOfFunction allOfFunction)
     return true;
 }
 
+template<typename T, typename U>
+void memcpySpan(std::span<T> destination, std::span<U> source)
+{
+    RELEASE_ASSERT(destination.size() == source.size());
+    static_assert(sizeof(T) == sizeof(U));
+    static_assert(!std::is_const_v<T>, "memcpySpan will not remove constness from destination");
+    memcpy(reinterpret_cast<void*>(destination.data()), reinterpret_cast<void*>(const_cast<std::remove_const_t<U>*>(source.data())), destination.size() * sizeof(T));
 }
 
+template<typename T>
+void memsetSpan(std::span<T> destination, uint8_t byte)
+{
+    static_assert(!std::is_const_v<T>, "memsetSpan will not remove constness from destination");
+    memset(reinterpret_cast<void*>(destination.data()), byte, destination.size() * sizeof(T));
+}
+
+template<typename T, typename U>
+std::span<T> spanCast(std::span<U> span)
```

Comment:
spanReinterpretCast() given what it actually does?

Reply:
Sounds good.